app/src/main/java/net/sourceforge/opencamera/sensorremote/RemoteRpcServer.java

```diff
@@ -0,0 +1,183 @@
+            outputStream.println(
+                    mRequestHandler.handleVideoStartRequest()
+            );
+        } else if (msg.equals(mConfig.getProperty("VIDEO_STOP_REQUEST"))) {
+            outputStream.println(
+                    mRequestHandler.handleVideoStopRequest()
+            );
+        } else if (msg.equals(mConfig.getProperty("GET_VIDEO_REQUEST"))) {
+            mRequestHandler.handleVideoGetRequest(outputStream);
+        }
+    }
+
+    @Override
+    public void run() {
+        // TODO: report hostname some other way
+        mContext.runOnUiThread(
+                () -> {
+                    try {
+                        new AlertDialog.Builder(mContext)
+                                .setTitle("Smartphone hostname")
+                                .setMessage("Hostname: " + getIPAddress())
+
+                                .setPositiveButton(android.R.string.yes, (dialog, which) -> {
+                                })
+                                .setIcon(android.R.drawable.ic_dialog_alert)
+                                .show();
+                    } catch (SocketException e) {
+                        e.printStackTrace();
+                    }
+
+                }
+        );
+
+        mIsExecuting.set(true);
+        if (MyDebug.LOG) {
+            Log.d(TAG, "waiting to accept connection from client...");
+        }
+        while (mIsExecuting.get()) {
+            try {
+                Socket clientSocket = mRpcSocket.accept();
+                clientSocket.setKeepAlive(true);
+                if (MyDebug.LOG) {
+                    Log.d(TAG, "accepted connection from client");
+                }
+                InputStream inputStream = clientSocket.getInputStream();
+                PrintStream outputStream = new PrintStream(clientSocket.getOutputStream());
+                BufferedOutputStream outputByte = new BufferedOutputStream(clientSocket.getOutputStream());
+                BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+
+                String inputLine;
+                while (!clientSocket.isClosed() && (inputLine = reader.readLine()) != null) {
+                    // Received new request from the client
+                    handleRequest(inputLine, outputStream, outputByte);
+                    outputStream.flush();
+                }
+                if (MyDebug.LOG) {
+                    Log.d(TAG, "closing connection to client");
+                }
+                outputByte.close();
+                outputStream.close();
+            } catch (SocketTimeoutException e) {
+               /* if (MyDebug.LOG) {
+                    Log.d(TAG, "socket timed out, waiting for new connection to client");
+                }*/
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        // Stopped accepting requests, close the server socket
+        try {
```

Comment:
Advice: enclose the whole processing loop into try/finally to `close` the socket in the `finally` block.